src/index.ts

```diff
@@ -0,0 +1,64 @@
+import * as pathToRegexp from 'path-to-regexp'
+import * as isPlainObject from 'is-plain-object'
+
+export interface Routes {
+ [pathPattern: string]: Routes | any
+}
+
+export interface Match {
+ path: string | null
+ pattern: string | null
+ value: any | null
+ params: string[] | null
+}
+
+const handlePath = (path: string, routes: Routes, parentPattern: string = ''): Match => {
```

Comment:
`parentPattern` has a default value, instead of the old-style reassignment line.